tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Graph.java

```diff
@@ -128,6 +136,77 @@ public GraphOperation operation(String name) {
     }
   }
 
+  /**
+   * Returns the operation (node in the Graph) with the provided name, or throws {@link IllegalArgumentException} if
+   * there isn't one.
+   *
+   * @param name name of the operation to look for
+   * @return operation in the graph with this name
+   * @throws IllegalArgumentException if no such operation exists in the Graph
+   */
+  public GraphOperation operationOrError(String name) {
```

Comment:
Cause that's a lot of boilerplate every time you want to get a for-sure existing op.  There's multiple places where it's used currently (graph walking and session, mostly), so we'd have utility functions for it anyways.

Reply:
A lot of the uses cases are places where we require the op to exist, and it should always exist, but if it somehow doesn't throwing the IllegalStateException is clearer and easier to debug than an NPE.

Reply:
As for the name, imo `OrError` sounded/read better than `OrException`, although now that I think about it `OrThrow` seems fine.  Or we could make these the default methods and use `OrNull` for the nullable ones.  Thoughts?

Reply:
I think I prefer `orThrow`, as that's more similar to the methods in the JDK - https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/Optional.html#orElseThrow(). It's worth noting that if we know the exception can't happen, then the one that returns null will likely be better optimised by the JIT compiler because exceptions get in the way.

Reply:
That makes sense.  Currently the only uses of the throwing variants are with user input, or in the graph walking functions.  The only case where the graph walking functions would have issues is if the graph is changed in another thread and the inputs were removed (which we don't even expose APIs for) so I'll go ahead and convert them to the non-throwing variant.